### ISSUE TITLE
seperating frontend and backend is not working properly

### ISSUE THREAD

**Issue by spring-bconnect** [aws-amplify/amplify-flutter#5272]
**### I am trying to setup seperate backend and frontend. I want to use Next js for website and flutter for android/ios from single backend amplify. When I try to follow the instructions it gives following error:**

**E/flutter ( 6589): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ConfigurationError {
E/flutter ( 6589):   "message": "No Cognito plugin config available"
E/flutter ( 6589): }**
E/flutter ( 6589): package:amplify_auth_cognito_dart/src/state/machines/configuration_state_machine.dart 69:7   ConfigurationStateMachine.onConfigure
E/flutter ( 6589): package:amplify_auth_cognito_dart/src/state/machines/configuration_state_machine.dart 50:15  ConfigurationStateMachine.resolve
E/flutter ( 6589): package:amplify_core/src/state_machine/state_machine.dart 262:59                             StateMachine._listenForEvents.<fn>.<fn>

Is there any way to achieve that but not able to do because of incomplete documentation. Can anyone please provide any solutions to achieve that.

**Comment by khatruong2009** [comment]
Hi @spring-bconnect, are you using the amplify-js library for the web portion of your application?

**Comment by spring-bconnect** [comment]
No. I have created a monorepo app and its working fine in local now but
when I try to deploy it, its failing. I am using yarn workspace in where
one app is nextjs for web and other is flutter for mobile apps. As per
documentation, structure for my app is:

Parent app
  Apps
    - app1(nextjs)
    - app2(flutter)
  Packages
    -shared
       -amplify
           -auth
           -data


Error displaying:

@amplify/backend is not found
Path to the amplify/data/resource is not recognized.


Can you provide a sample example for monorepo setup and deployment.
Documentation is incomplete.

Main issue: path to resource is not found in front end apps

**Comment by spring-bconnect** [comment]
<img width="1440" alt="Screenshot 2024-08-13 at 9 29 54 AM" src="https://github.com/user-attachments/assets/d2bfe17f-7b18-4515-a469-fc7e7bd078a5">

**Comment by spring-bconnect** [comment]
<img width="1440" alt="Screenshot 2024-08-13 at 4 14 21 PM" src="https://github.com/user-attachments/assets/3db33ef7-5aa2-4c81-b5a2-d14d960b1b6d">


How to generate this amplify folder if its separate frontend and backend?? As per the documentation given : 
[]( https://docs.amplify.aws/nextjs/deploy-and-host/fullstack-branching/mono-and-multi-repos/)

Only amplify_outputs.json is created?

Can you provide next steps so that build does not fail??

**Comment by khatruong2009** [comment]
Hi @spring-bconnect, I see that you've run the first command to get the `amplify_outputs.json` file. Can you also run `npx ampx generate outputs --app-id <your-backend-amplify-app-id> --branch main --format dart --out-dir lib` command so that you get the amplify_outputs.dart file as well for the dart side as well?

**Comment by spring-bconnect** [comment]
<img width="1626" alt="Screenshot 2024-08-14 at 2 41 32 PM" src="https://github.com/user-attachments/assets/7749538e-2e3b-43a6-8fbf-960686d92ffb">

I did the same for dart amplify outputs generation. Getting the same when I try to deploy next js app

**Comment by spring-bconnect** [comment]
Amplify folder is outside the frontend apps in monorepo structure.. in shared backend.. As per error it constructs:

**Amplify Backend not found**

What setup is required so that it points to amplify backend generated and hosted separately on server.

**Comment by salihgueler** [comment]
Hello @spring-bconnect, thanks for reaching out. With Monorepo support, you should not need any amplify folder as it is in a separate place. So that part should not be a problem. 

For your application to understand what kind of configuration you have, it should generate **amplify_outputs.json** for Next.js application and **amplify_outputs.dart** for the Flutter application.

For generating the **amplify_outputs.dart**, you can run the following command in the amplify.yaml file that you can find in the build settings.

```yaml
version: 1
backend:
  phases:
    build:
      commands:
        - npm ci --cache .npm --prefer-offline
        - npx ampx generate outputs --branch $AWS_BRANCH --app-id $AWS_APP_ID --format dart --out-dir lib
```

Also, you did the setup as monorepo, right? So the amplify folder is known by the console.
